src/server/v2.0/harbor/handler/handler.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/go-openapi/errors"
+	"github.com/goharbor/harbor/src/server/middleware"
+	"github.com/goharbor/harbor/src/server/v2.0/harbor/handler/artifact"
+	"github.com/goharbor/harbor/src/server/v2.0/harbor/restapi"
+	"github.com/justinas/alice"
+)
+
+// New returns http handler for API V2.0
+func New() http.Handler {
+	h, api, err := restapi.HandlerAPI(restapi.Config{
+		InnerMiddleware: innerMiddleware,
```

Comment:
The request ID middleware should cover all requests: v1 APIs, v2 APIs and registry APIs

Reply:
I think this out of scope of this pr.